### ISSUE TITLE
Kernel panic on to_msgpack

### ISSUE THREAD

**Issue by justinwiley** [msgpack/msgpack-ruby#77]
I'm using Ruby 2.2.2 on Mac 10.10.4, and the stuff-classifier gem, which pulls in msgpack 0.6.1.

I've trained the classifier on a large list of words, resulting in a hash 5mb hash structure like so:

```ruby
{"aardvark"=>{:public=>1},
 "aardwolf"=>{:public=>1},
 "aaron"=>{:public=>1, :confidential=>3},
 "aback"=>{:public=>1},
 "abacu"=>{:public=>1},
 "abaft"=>{:public=>1},
```

The classifier tries to pack it, and a kernel panic ensues.  I've tried recreating this by creating another large hash, but it serializes without issue.

[https://gist.github.com/justinwiley/9a6c59000949a1fcac3a](https://gist.github.com/justinwiley/9a6c59000949a1fcac3a)

**Comment by frsyuki** [comment]
it could be related to #75 which is reported today and merged today. we'll release new version soon.

**Comment by nurse** [comment]
@frsyuki he says it is Ruby 2.2.2 and the backtrace also says that.
#75's case doesn't affect Ruby 2.2 because of its around ifdef.

**Comment by frsyuki** [comment]
msgpack_packer_write_bin_header fix affects Ruby 2.2.2.

**Comment by tagomoris** [comment]
msgpack 0.6.2 is now out with fix for #75.
@justinwiley Can you try it?

**Comment by tagomoris** [comment]
Let me close this issue.
@justinwiley Please reopen this issue if you still have problem on master HEAD.
